cmd/minikube/cmd/config/config.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"io"
+
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+type configFile interface {
+	io.ReadWriter
+}
+
+type setFn func(string, string) error
+type MinikubeConfig map[string]interface{}
+
+type Setting struct {
+	name        string
+	set         func(MinikubeConfig, string, string) error
+	validations []setFn
+	callbacks   []setFn
+}
+
+// These are all the settings that are configurable
+// and their validation and callback fn run on Set
+var settings []Setting = []Setting{
+	{
+		name:        "vm-driver",
+		set:         SetString,
+		validations: []setFn{IsValidDriver},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	{
+		name:        "v",
+		set:         SetInt,
+		validations: []setFn{IsPositive},
+	},
+	{
+		name:        "cpus",
+		set:         SetInt,
+		validations: []setFn{IsPositive},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	{
+		name:        "disk-size",
+		set:         SetString,
+		validations: []setFn{IsValidDiskSize},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	{
+		name:        "host-only-cidr",
+		set:         SetString,
+		validations: []setFn{IsValidCIDR},
+	},
+	{
+		name:        "memory",
+		set:         SetInt,
+		validations: []setFn{IsPositive},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	{
+		name: "show-libmachine-logs",
+		set:  SetBool,
+	},
+	{
+		name:        "log_dir",
+		set:         SetString,
+		validations: []setFn{IsValidPath},
+	},
+	{
+		name: "kubernetes-version",
+		set:  SetString,
+	},
+}
+
+var ConfigCmd = &cobra.Command{
+	Use:   "config SUBCOMMAND [flags]",
+	Short: "Modify minikube config",
+	Long:  `config modifies minikube config files using subcommands like "minikube config set vm-driver kvm"`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+// Reads in the JSON minikube config
+func ReadConfig() MinikubeConfig {
+	f, err := os.Open(constants.ConfigFile)
```

Comment:
Sorry, one last nit. It looks like the error handling in this function is a bit off. glog.Errorf doesn't return anything, it just logs the error. It looks like you could end up returning an unitialized map here if Open throws an error, or decode throws an error.